ha/util/message_bus.py

```diff
@@ -0,0 +1,166 @@
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import json
+from typing import Callable
+from threading import Thread
+from cortx.utils.message_bus import MessageBusAdmin
+from cortx.utils.message_bus.error import MessageBusError
+from cortx.utils.message_bus import MessageProducer
+from cortx.utils.message_bus import MessageConsumer
+import ctypes
+
+class MessageBusProducer:
+    PRODUCER_METHOD = "sync"
+
+    def __init__(self, producer_id: str, message_type: str, partitions: int):
+        """
+        Register message types with message bus.
+
+        Args:
+            producer_id (str): producer id.
+            message_types (str): Message type.
+            partitions (int, optional): No. of partitions. Defaults to 1.
+
+        Raises:
+            MessageBusError: Message bus error.
+        """
+        self.producer = MessageProducer(producer_id=producer_id, message_type=message_type, method=MessageBusProducer.PRODUCER_METHOD)
+
+    def publish(self, message: any):
+        """
+        Produce message to message bus.
+
+        Args:
+            message (dict): Message.
+        """
+        if isinstance(message, dict):
+            self.producer.send([json.dumps(message)])
+        elif isinstance(message, str):
+            self.producer.send([message])
+        else:
+            raise Exception(f"Invalid type of message {message}")
+
+class MessageBusConsumer:
+
+    def __init__(self, consumer_id: int, consumer_group: str, message_type: str,
+                callback: Callable, auto_ack: bool, offset: str):
+        """
+        Initalize consumer.
+
+        Args:
+            consumer_id (int): Consumer ID.
+            consumer_group (str): Consumer Group.
+            message_type (list): Message Type.
+            callback (Callable): function to get message.
+            auto_ack (bool, optional): Check auto ack. Defaults to False.
+            offset (str, optional): Offset for messages. Defaults to "earliest".
+        """
+        self.callback = callback
+        self.consumer = MessageConsumer(consumer_id=str(consumer_id),
+                        consumer_group=consumer_group,
+                        message_types=[message_type],
+                        auto_ack=auto_ack, offset=offset)
+        self.consumer_thread = Thread(target=self.run)
+        self.consumer_thread.setDaemon(True)
+        self.stop_thread = False
+
+    def run(self):
+        """
+        Overloaded of Thread.
+        Stop thread by
+        1. Return True from callback passed during initialization.
+        2. Close main thread as deamon this thread will be closed after main.
+
+        As self.consumer.receive(timeout=0) is block call t1.join() will not stop thread.
+        """
```

Comment:
Are this 2 way stop enough
is there need to stop thread from main thread like t1.stop()

Reply:
Resolved as per discussion 
added multiple case